### PR TITLE
Bump version from 3.7.0 to 3.8.0

### PR DESCRIPTION
**Pull request by rvl**
- Bump the cardano-addresses version for a new release.
- Add release notes
- Improve the text of `cardano-address key walletid --help`

### CODE REVIEW

File: command-line/lib/Command/Key/WalletId.hs [hunk 1]
```diff
@@ -37,13 +37,17 @@ data Cmd = WalletId
 mod :: (Cmd -> parent) -> Mod CommandFields parent
 mod liftCmd = command "walletid" $
     info (helper <*> fmap liftCmd parser) $ mempty
-        <> progDesc "Get the wallet id from a given key"
+        <> progDesc "Shows the cardano-wallet WalletId for a given key"
```
Comment (jonathanknowles) [inline command-line/lib/Command/Key/WalletId.hs:40]:
```suggestion
        <> progDesc "Shows the cardano-wallet wallet ID for a given key"
```

File: command-line/lib/Command/Key/WalletId.hs [hunk 2]
```diff
@@ -37,13 +37,17 @@ data Cmd = WalletId
...
         <> footerDoc (Just $ string $ mconcat
-            [ "The key is read from stdin and"
-            , "is assumed to be either extended root or extended account key."
-            , "Both extended private and extended public keys are accepted and"
-            , "the corresponding duals give rise to the same wallet id."
-            , "Wallet id is hex-encoded Blake2b-160 hash applied on an extended public key."
+            [ "A WalletId is a 40-digit hexadecimal string derived "
```
Comment (jonathanknowles) [inline command-line/lib/Command/Key/WalletId.hs:42]:
```suggestion
            [ "A wallet ID is a 40-digit hexadecimal string derived "
```

File: command-line/lib/Command/Key/WalletId.hs [hunk 3]
```diff
@@ -37,13 +37,17 @@ data Cmd = WalletId
...
+            , "from the wallet’s mnemonic. It is used by the cardano-wallet "
+            , "server to refer to specific wallets.\n\n"
+            , "This key can be either an extended root key "
+            , "(full multi-account wallets), or an extended account key "
+            , "(single-account wallets).\n\n"
+            , "Private or public extended keys are accepted -- they will have "
+            , "the same WalletId.\n\n"
```
Comment (jonathanknowles) [inline command-line/lib/Command/Key/WalletId.hs:49]:
```suggestion
            , "the same wallet ID.\n\n"
```

**Review by jonathanknowles — APPROVED** [review]
Hi @rvl 

Seems good to me.

Though what do you think about using "wallet ID" instead of "WalletId" in human-facing documentation?

We already use language like "wallet mnemonic" rather than "WalletMnemonic".

File: command-line/lib/Command/Key/WalletId.hs (hunk 1 above)
Comment (rvl) [reply command-line/lib/Command/Key/WalletId.hs]: I was trying to suggest that WalletId is not just any ID, but a specific type of ID.
Perhaps not then... suggestions applied.

**Pull request merged**